add a new firmware

jtolio opened the pull request:
just wanted to get the word out!

jtolio committed:
add a new firmware

just wanted to get the word out!

Szybet commented:
Woah nice work!

As the current section is my personal list, do you agree that after I accept this pull request I will move it to another section of "Contributed / Other" firmwares and write a few more things about it (Default SQFMI firmware based, only arduino ide)?

Offtopic:
Wouldn't you want to convert it to a library so other firmwares could use it? Idk if many people are out there who would want to switch to a entire other firmware for just calendar features (As inkwatchy already supports calendar, not so many features or so good UI but it's there) - It would also enable to use with platformio for example, more devices support (Yatchy for example). If you do not like this idea, it's completely fine, it's just my thinking, anyway, good job! It's always good to see some innovation in the space

jtolio commented:
It's not only Arduino IDE - it works with Arduino CLI also. I also ended up throwing out most of the SQFMI firmware code. I agree that I haven't made a PlatformIO configuration yet. I was just reading your list of why you should always use Platform IO and the Arduino CLI project addresses those concerns. But all the same, I will look into making a library definition so it can be used by other projects like Platform IO! Thanks for the consideration!

Szybet commented:
> As the current section is my personal list, do you agree that after I accept this pull request I will move it to another section of "Contributed / Other" firmwares and write a few more things about it (Default SQFMI firmware based, only arduino ide)?

I assume yes then, feel free to create an issue if the final look doesn't fit you

I hope the library will be async (as in not a big loop which locks the rest up) and things like display definition, button function request could be provided by the firmware (as a digitalRead will not do)
